addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeAggregateDataPolling.java

```diff
@@ -0,0 +1,88 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdatePrivateApi;
+import org.openhab.binding.solaredge.internal.command.AggregateDataUpdatePublicApi;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.openhab.binding.solaredge.internal.model.AggregatePeriod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Polling worker class. This is responsible for periodic polling of sensor data.
+ *
+ * @author Alexander Friese - initial contribution
+ */
+public class SolarEdgeAggregateDataPolling implements Runnable {
+    /**
+     * Logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Handler for delegation to callbacks.
+     */
+    private final SolarEdgeHandler handler;
+
+    /**
+     * Constructor.
+     *
+     * @param handler handler which handles results of polling
+     */
+    public SolarEdgeAggregateDataPolling(SolarEdgeHandler handler) {
+        this.handler = handler;
+    }
+
+    /**
+     * Poll the SolarEdge Webservice one time per call.
+     */
+    @Override
+    public void run() {
+        if (handler.getWebInterface() != null) {
+            // if no meter is present all data will be fetched by the 'LiveDataUpdateMeterless'
+            if (handler.getConfiguration().isMeterInstalled()) {
+                logger.debug("polling SolarEdge aggregate data {}", handler.getConfiguration());
+
+                List<SolarEdgeCommand> commands = new ArrayList<>();
+
+                if (handler.getConfiguration().isUsePrivateApi()) {
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.DAY));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.WEEK));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.MONTH));
+                    commands.add(new AggregateDataUpdatePrivateApi(handler, AggregatePeriod.YEAR));
+                } else {
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.DAY));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.WEEK));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.MONTH));
+                    commands.add(new AggregateDataUpdatePublicApi(handler, AggregatePeriod.YEAR));
+                }
+
+                try {
+                    for (SolarEdgeCommand command : commands) {
+                        handler.getWebInterface().executeCommand(command);
+                        // we must not flood the API with requests, otherwise we will get an exception
+                        Thread.sleep(5000);
+                    }
+                } catch (RuntimeException e) {
```

Comment:
You should avoid to catch RuntimeExceptions - better catch (and throw) more specific ones.

Reply:
I had in the past a few NPEs due to unexpected API responses. I expected data to be there but partly got null values which caused NPEs in further processing.
However the binding stopped working silently which means without any exception logged or anything else. This was the reason why I added this.

Reply:
I would then expect the `executeCommand` to already catch the NPEs internally and not let them bubble up. 

> However the binding stopped working silently

This should meanwhile be addressed by https://github.com/eclipse/smarthome/pull/4772 - if a job dies because of unexpected exceptions, you will find an error in the log.

Reply:
Ok, I think in that case I could remove the try-catch blocks completely. Because currently I do not know any specific exception which needs to be handled. If it gets logged by the Threadpool I am happy.

Reply:
I removed the exception handling completely.